HotelReservation/Business/Services/RoomsService.cs

```diff
@@ -94,29 +89,38 @@ public async Task UpdateRoom(Guid roomId, string userId,RoomModel room)
                 await _roomRepository.UpdateAsync(roomEntity);
             }
         }
-        public async Task<PageInfo<RoomModel>> GetRoomsPage(Guid hotelId,Pagination hotelPagination)
+        public async Task<PageInfo<RoomModel>> GetRoomsPage(Guid hotelId,DateTime checkInDate,DateTime checkOutDate, Pagination roomPagination)
         {
             var hotelEntity = await _hotelRepository.GetAsync(hotelId);
             if (hotelEntity == null)
             {
                 _logger.LogError($"hotel with {hotelId} id not exists");
                 throw new NotFoundException($"hotel with {hotelId} id not exists");
             }
-            var rooms = _roomMapper.Map<IEnumerable<RoomModel>>(_roomRepository.GetRoomsPageFromHotel(hotelPagination.PageNumber,
-                hotelPagination.PageSize,hotelId));
-            var numberOfRooms = await _roomRepository.GetRoomsCount(hotelId);
 
-            int  numberOfPages = numberOfRooms / hotelPagination.PageSize;
-            if (numberOfRooms % hotelPagination.PageSize != 0)
-            {
-                numberOfPages++;
-            }
-            var roomPageInfo = new PageInfo<RoomModel>
+            var filteredRooms = new List<RoomEntity>();
+            if (hotelEntity.Rooms != null)
             {
-                Items = rooms, NumberOfItems = numberOfRooms, NumberOfPages = numberOfPages
-            };
+                foreach (var room in hotelEntity.Rooms)
+                {
+                    if (room.Orders != null && room.Orders.Count!=0)
+                    {
+                        if (room.Orders.All(order => !(checkInDate > order.StartDate && checkInDate < order.EndDate || checkOutDate > order.StartDate && checkOutDate < order.EndDate
+                            || order.StartDate > checkInDate && order.StartDate < checkOutDate || order.EndDate > checkInDate && order.EndDate < checkOutDate)))
```

Comment:
If you use this complex condition in several places it would be better to create a function for this.

Reply:
extracted to function